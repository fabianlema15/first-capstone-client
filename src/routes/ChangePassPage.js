import React from 'react';
import GenericContext from '../contexts/GenericContext'
import ChangePassContent from '../components/User/ChangePass'
import AuthApiService from '../services/auth-api-service';
import TokenService from '../services/token-service'

class ChangePassPage extends React.Component{
  static defaultProps = {
    ROUTE : 'logins'
  }

  state = {
    error: null,
    changed: false,
  };

  setError = error => {
    this.setState({ error: error.error })
  }

  clearError = () => {
    this.setState({ error: null })
  }

  changePassword = (e) => {
    e.preventDefault();
    const { password, password_rep } = e.target;
    if (password.value !== password_rep.value)
      this.setError({error: 'Passwords do not match!'})
    else
      AuthApiService.changePassword({id: TokenService.decodeUser().user_id, password: password.value})
        .then(result => this.setState({changed: true}))
        .catch(this.setError)
  }

  render() {
    const value = {
      clearError: this.clearError,
      error: this.state.error,
      changePassword: this.changePassword,
      setError: this.setError,
      objName: 'Change Password',
      changed: this.state.changed
    }

    return (
      <GenericContext.Provider value={value}>
        <ChangePassContent />
      </GenericContext.Provider>
    )
  }
}

export default ChangePassPage;
