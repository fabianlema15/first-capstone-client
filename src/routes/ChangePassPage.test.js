import React from 'react';
import ReactDOM from 'react-dom';
import ChangePassPage from './ChangePassPage'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';

describe('<ChangePassPage />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
      <ChangePassPage>
      </ChangePassPage>
    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Renders ChangePassPage as expected', () => {
	    const tree = renderer.create(<BrowserRouter>
        <ChangePassPage>
        </ChangePassPage>
      </BrowserRouter>).toJSON();
	    expect(tree).toMatchSnapshot();
	  });

})
