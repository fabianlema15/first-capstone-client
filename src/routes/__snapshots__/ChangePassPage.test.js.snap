// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChangePassPage /> Renders ChangePassPage as expected 1`] = `
<div>
  <nav>
    <div
      className="back-nav"
    >
      <a
        href="/menu"
        onClick={[Function]}
      >
        Back
      </a>
    </div>
    <div
      className="name-nav"
    >
      <h2>
        Change Pas
        s
      </h2>
    </div>
  </nav>
  <main
    className="content-with-nav"
  >
    <section>
      <form
        onSubmit={[Function]}
      >
        <h3>
          Change Password
        </h3>
        <fieldset>
          <label
            htmlFor="password"
          >
            Password
            <input
              className="Input "
              id="password"
              onChange={[Function]}
              required={true}
              type="password"
              value=""
            />
          </label>
        </fieldset>
        <fieldset>
          <label
            htmlFor="password_rep"
          >
            Repeat Password
            <input
              className="Input "
              id="password_rep"
              onChange={[Function]}
              required={true}
              type="password"
              value=""
            />
          </label>
        </fieldset>
        <button
          className="green"
          type="submit"
        >
          Change Password
        </button>
      </form>
    </section>
  </main>
</div>
`;
